booking.py

```diff
@@ -107,6 +108,106 @@ def add_tagged_friends(friends):
+            friend1: /friend1_base_dir/root_file3/folder/ntuple
+            friend2: /friend2_base_dir/root_file3/folder/ntuple
+        (...)
+
+    Args:
+        dataset_name (str): Name of the dataset
+        file_names (list): List containing the names of the .root
+            files
+        folder (str): Name of the TDirectoryFile in each .root file
+        files_base_directory (str): Path to the files base directory (directories)
+        friends_base_directories (str, list): List of paths to
+            the friends base directory (directories)
+
+    Returns:
+        dataset (Dataset): Dataset object containing TTrees
+    """
+    def get_quantities_per_variation(path_to_root_file):
+        root_file = TFile(path_to_root_file)
+        if root_file.IsZombie():
+            logger.fatal('File {} does not exist, abort'.format(path_to_root_file))
+            raise FileNotFoundError
+        quantities_per_vars={}
+        quantities_with_variations = root_file.Get("ntuple").GetListOfLeaves() 
+        for qwv in quantities_with_variations:
+            qwv_name = qwv.GetName()
+            if "__" in qwv_name:
+                quantity,var = qwv_name.split("__")
+                if var not in quantities_per_vars.keys():
+                    quantities_per_vars[var] = []
+                quantities_per_vars[var].append(quantity)
+        root_file.Close()
+        return quantities_per_vars
+
+
+    def get_full_tree_name(folder, path_to_root_file, tree_name):
+        root_file = TFile(path_to_root_file)
+        if root_file.IsZombie():
+            logger.fatal('File {} does not exist, abort'.format(path_to_root_file))
+            raise FileNotFoundError
+        root_file.Close()
+        full_tree_name = tree_name
+        return full_tree_name
+
+    def add_tagged_friends(friends):
+        """ Tag friends with the name of the different directories
+        in the artus name scheme, e.g.:
+        /common_path/MELA/ntuple -> tag: MELA
+        /common_path/SVFit/ntuple -> tag: SVFit
+        Since when we compare two ntuples (with full path) only one
+        directory changes in this scheme (see MELA vs SVFit), we
+        create a list called 'tags' with these two strings; then we
+        assign this string to friend.tag, if it's None
+        """
+        for f1, f2 in itertools.combinations(friends, 2):
+            l1 = f1.path.split('/')
+            l2 = f2.path.split('/')
+            tags = list(set(l1).symmetric_difference(set(l2)))
+            if tags:
+                for t in tags:
+                    if t in l1 and f1.tag is None:
+                        f1.tag = t
+                    elif t in l2 and f2.tag is None:
+                        f2.tag = t
+        return friends
+
+    root_files = []
+    for f in file_names:
+        for g in os.listdir(os.path.join(files_base_directory, f)):
+            print(os.path.join(files_base_directory, f, g))
```

Comment:
I don't think the `print` statement is necessary here (will likely produce too much output for standard use). It is probably the best solution to use the logger for this if we really need the output.
```suggestion
            logger.debug(os.path.join(files_base_directory, f, g))
```